### ISSUE TITLE
[HOLD for payment 2023-08-21] Web - Cannot remove contact method if it has an error

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23180]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings
2. Go to Profile
3. Go to Contact method
4. Add a new contact method with a long email
5. Observe that an error occurs and the contact method can't be added
6. Attempt to remove the contact method
7. Observe that the contact method does not disappear

## Expected Result:
The contact method should be removed

## Actual Result:
The contact method remains

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.42.25
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/4b12ca98-2698-40a5-a501-c236ad3131a2

https://github.com/Expensify/App/assets/93399543/e1a9c022-d64c-4bee-ace1-c86016ac2d4a

**Expensify/Expensify Issue URL:**
**Issue reported by:** @samh-nl
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689696657812329
 
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01002b21b83bc2bacc</li>
        <li>Upwork Job ID: 1686053289250795520</li>
        <li>Last Price Increase: 2023-07-31</li>
    </ul>
</details>

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Cannot remove contact method if it has an error

### What is the root cause of that problem?
We call `User.deleteContactMethod` which starts an API call, but, if contact method had errors when added, backend does not have it and retuns an error, thus Onyx uses `failureData`, keeping the contact method:
https://github.com/Expensify/App/blob/f2259369ed001c77cd6670a8416217f431eb523a/src/libs/actions/User.js#L224-L279

### What changes do you think we should make in order to solve the problem?
Before starting the API call, we should verify if the contact method had errors when added, thus we'll only remove it from Onyx, as it's not present in the back-end:
```js
// If contact method was not added, only remove it locally
if (!_.isEmpty(lodashGet(loginList, [contactMethod, 'errorFields'], {}).addedLogin)) {
    Onyx.update(successData);
} else {
    API.write(
        'DeleteContactMethod',
        {
            partnerUserID: contactMethod,
        },
        {optimisticData, successData, failureData},
    );
}
```

### What alternative solutions did you explore?
In my Proposal I verified if `addedLogin` is present. As an alternative, we could prevent back-end calls for any error using `if (!_.isEmpty(lodashGet(loginList, [contactMethod, 'errorFields'], {}) {`.

**Comment by cristipaval** [comment]
Thanks for pinging me @bernhardoj! You're right, we want the backend to handle the contact method deletion even if the contact is not claimed by the account.

@stephanieelliott, I'll check soon to see if this should be internal.

**Comment by stephanieelliott** [comment]
Hey @cristipaval, gentle bump and a reminder to check if this should be handled internally 😊

**Comment by cristipaval** [comment]
hey @stephanieelliott! I just checked the root cause of this issue and I need to start an internal discussion with the other engineers.

**Comment by cristipaval** [comment]
Started a discussion in Slack [here](https://expensify.slack.com/archives/C03TQ48KC/p1690482472889119)

**Comment by cristipaval** [comment]
FYI: The root cause of the issue was that the email passed the front-end validation when added as a new contact method, but it doesn't pass the backend validation. The fix for this is to make the front-end validation consistent with the backend email validation. [PR](https://github.com/Expensify/App/pull/23993) to fix the issue is up.

**Comment by stephanieelliott** [comment]
PR is undergoing internal review!

**Comment by stephanieelliott** [comment]
PR waiting on your review @rushatgabhane

**Comment by rushatgabhane** [comment]
created a manual request for $1k https://staging.new.expensify.com/r/510471181112893

**Comment by stephanieelliott** [comment]
Hey @rushatgabhane can you complete the BZ checklist when you have a second?

**Comment by JmillsExpensify** [comment]
@stephanieelliott Can you also summarize payouts when you get a chance? (Sorry if I missed if above!)

**Comment by stephanieelliott** [comment]
@rushatgabhane, another reminder to complete the BZ checklist, please! 

@JmillsExpensify I'll post a payment summary below, however I think we typically wait until the BZ checklist is completed before issuing payment. I'll keep the below summary updated though! 

Summarizing payment on this issue: 
 - Issue reporter: @samh-nl $250 (pay via Upwork -- offer extended, awaiting acceptance)
 - Contributor: N/A [internal] 
 - Contributor+: @rushatgabhane $1000 (to pay in Newdot - **on hold until BZ checklist is completed**)
 
 Upwork job is [here](https://www.upwork.com/jobs/~01002b21b83bc2bacc), no bonus is included on final payout

**Comment by stephanieelliott** [comment]
Thanks @samh-nl, I've issued payment. 

@rushatgabhane another bump to complete the BZ checklist please

**Comment by JmillsExpensify** [comment]
Reviewed the details for @rushatgabhane. $1,000 payment is approved via NewDot based on BZ summary above.
